### PR TITLE
Added setup instructions and script. Local setup fixes.

### PR DESCRIPTION
**Pull request by benubois**
Hi @kfogel,

This adds some instructions and a script for getting `arcdata` running locally. The `bin/setup` script is pretty much the default setup script that rails ships with.

I also made a few code changes including:
- Updated a dependency that was giving me errors (therubyracer).
- Updated `db/seeds.rb` to run without errors.
- Updated `config/initializers/connect.rb` to create public/private key pair in memory. Not really sure how this is used, but it will satisfy [openid-connect-engine](https://github.com/jlaxson/openid-connect-engine) so the app will boot.

You can create a new user by running `rails console` and then:

``` ruby
Roster::Person.create(chapter: Roster::Chapter.first, email: "example@example.com", username: "example@example.com", password: "password")
```

~~However, this doesn't seem to let me login and I haven't spend enough time with the `Roster::SessionsController` to find out why.~~ Looks like you need to specify a `username` when creating a `Person`. The create example above has been updated to include this.

Please let me know if you have any questions.

Thanks!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,18 @@
...
\ No newline at end of file
+# ARCData
+
+ARCData is a tool for managing some Red Cross Disaster Services processes, such as managing DAT shifts.
+
+## Setup
+
+ARCData is a Ruby on Rails app and depends on:
+
+- [PostgreSQL XX](http://www.postgresql.org/download/)
```
Comment (benubois) [inline README.md]: The version of Postgres used in production should go here.

File: .ruby-version [hunk 2]
```diff
@@ -0,0 +1 @@
+2.1.2
```
Comment (benubois) [inline .ruby-version]: The helps ruby version managers, like [rbenv](https://github.com/rbenv/rbenv), to use the correct version of ruby

**Comment by kfogel** [comment]
Heading off to help a friend move tonight, but will review and merge when
get home.  Thank you, Ben!  Also, we have those screenshots now; I will
send then (but also deploy and compare myself).

**Comment by benubois** [comment]
Looks like the failing test is fixed in #59. Here's a screenshot of what it looks like for me running locally.

screen shot 2016-02-04 at 3 28 28 pm

**Comment by benubois** [comment]
[dummy_data](https://github.com/goncalossilva/dummy_data) is an example of something that generates data based on the schema and model relationships. However it does not work with this project. I think it would be worth trying to get this to work or looking into other options like this.

There are also more manual options like using [faker](https://github.com/stympy/faker) and creating models by hand.

For instance:

```ruby
chapter = Roster::Chapter.create(
  name: Faker::University.name, 
  short_name: Faker::Hacker.abbreviation, 
  code: Faker::Address.zip_code, 
  time_zone_raw: Faker::Address.time_zone,
)

10.times.do |n|
  Roster::Person.create(chapter: chapter, email: Faker::Internet.email, password: Faker::Internet.password)
end
```

This could be tough with the number of models and relationships in this site.

**Comment by kfogel** [comment]
Okay, had a chance to look in more detail now.  I'll save the merge for tomorrow when I'm more awake, but this is great progress; thanks @benubois!  The failing test is clearly just the same thing pull request #59 fixes; no worries there.  I think your suggested route(s) for creating dummy data (as per issue #67) are the right direction to go.

The $64,000 question is what do you see when you log in from that login screen?  We've gotten things to the point of showing that screen before too, it's just that after login (I think we created an admin user by hand, in order to log in) a lot of the site is still missing.  We hypothesized that for a lot of the missing pages or page areas, the reason it's missing is the absence of data that would normally cause it to be generated.  To test that hypothesis, I think the next step is to log in and do a systematic comparison of our still-hobbled dev sandboxes with the screenshots from the live site; will try that tomorrow.

More soon.  CC'ing @OhMcGoo (Jim McGowan).

**Comment by cecilia-donnelly** [comment]
Hey @benubois, thanks so much!  I've now also logged in locally, thanks to your help.  As you predicted, I got some errors running the setup script, but I debugged those and got it running over here.

I can log in if I create a user, but the homepage after signin still gives me an error on master.  If I check out the [`deployment-experiments` branch](https://github.com/OpenTechStrategies/arcdata/commits/deployment_experiments) then the change in [this commit](https://github.com/OpenTechStrategies/arcdata/commit/2f25ea881fd1780ae3408f4b1fac31257d318dbe) handles the error and I get what you see in the screenshot below.  

@kfogel I vote that we merge this PR.  The setup script is super useful.  Thanks again, Ben.


screenshot from 2016-02-05 17-32-06

**Comment by cecilia-donnelly** [comment]
Merged this PR to branch `dev-setup`.  I'll be using that as the main space for fixing #73 and getting this app fully deployable locally.  I tagged you in the issue, @benubois, so you should see further conversation happening there.

**Comment by benubois** [comment]
> The $64,000 question is what do you see when you log in from that login screen?

@kfogel, I get an error, probably due to missing data.

> I tagged you in the issue, @benubois, so you should see further conversation happening there.

@cecilia-donnelly ok, thanks! Let me know if you need anything else.

**Comment by frankduncan** [comment]
4 years later, it looks like most of the stuff in bin/setup here is handled by `rake:bootstrap`, so it's not really valid anymore.  Closing to clean up the repository

**Pull request closed without merging**